external/vcm/tests/test_derived_mapping.py

```diff
@@ -172,3 +172,24 @@ def test_is_land():
 def test_is_sea_ice():
     derived_state = DerivedMapping(ds_sfc)
     np.testing.assert_array_almost_equal(derived_state["is_sea_ice"], [0.0, 0.0, 1.0])
+
+
+@pytest.mark.parametrize(
+    "dependency_map, derived_vars, reqs",
+    [
+        ({"c": ["d"]}, ["a"], []),
+        ({"a": ["b"], "c": ["d"]}, ["a"], ["b"]),
+        ({"a": ["b"], "c": ["d"]}, ["a", "c"], ["b", "d"]),
+        ({"a": ["b"], "b": ["c"], "c": ["d"]}, ["a"], ["b", "c", "d"]),
```

Comment:
I think this would fail due to the `new_deps` bug.
`({"a": ["b"], "c": ["d"], "b": ["e"]}, ["a", "c"], ["b", "d", "e"]),`

Reply:
Thanks, good catch- fixed now and added your test.